### ISSUE TITLE
constructor default parameters are reused for case class apply

### ISSUE THREAD

**Issue by bishabosha** [lampepfl/dotty#13324]
A case class apply method does not have default parameters generated, instead the constructor default parameters are reused, I believe that this is undocumented - causing https://github.com/scala/bug/issues/12409

## Compiler version

3.0.1

## Minimized code

```Scala
case class CaseClassDefault(value: Int = 1)

val foo = Scala3CaseClass.apply()
```

## Output

```scala
val foo: CaseClassDefault = CaseClassDefault.apply(CaseClassDefault.`<init>$default$1`)
```

## Expectation

```scala
val foo: CaseClassDefault = CaseClassDefault.apply(CaseClassDefault.`apply$default$1`)
```

## Notes

extract from generated code:
```scala
final lazy module val CaseClassDefault: CaseClassDefault$ = 
  new CaseClassDefault$()
final module class CaseClassDefault$() extends AnyRef() { 
  this: CaseClassDefault.type =>
  def apply(value: Int): CaseClassDefault = new CaseClassDefault(value)
  def unapply(x$1: CaseClassDefault): CaseClassDefault = x$1
  override def toString: String = "CaseClassDefault"
  def $lessinit$greater$default$1: Int @uncheckedVariance = 1
}
```

**Comment by bishabosha** [comment]
this looks by design in Applications, but it is an incompatibility
https://github.com/lampepfl/dotty/blob/45ce129a2c166a6fbb64dc096fe50435f1f77f96/compiler/src/dotty/tools/dotc/typer/Applications.scala#L223

**Comment by odersky** [comment]
Yes, synthesized apply methods take the default getters of their constructors. If we did not want to do that, we would either make `apply` methods useless, or have to duplicate all default getters. I believe it is better this way. Can Tasty reader be adapted to deal with it?

**Comment by bishabosha** [comment]
> Can Tasty reader be adapted to deal with it?

It should be possible, I am looking into it

**Comment by bishabosha** [comment]
PR open here https://github.com/scala/scala/pull/9730

**Comment by odersky** [comment]
Great! @bishabosha Can I leave the issue here for to you to close once the scala/scala side is resolved?

**Comment by TheElectronWill** [comment]
scala/scala#9730 has been closed, so this issue should be closed too?

**Comment by bishabosha** [comment]
perhaps an extra note in the documentation about the change to default parameter encoding is required
